### ISSUE TITLE
Error building schema if '#' not in ref

### ISSUE THREAD

**Issue by danielgtaylor** [pb33f/libopenapi#37]
The official Open API spec shows an [example of a relative schema document reference](https://spec.openapis.org/oas/latest.html#relative-schema-document-example) without an explicit `#`:

```yaml
{
  "$ref": "Pet.json"
}
```

When I try this with libopenapi it fails to build the schema:

```sh
$ go run ./bug
panic: schema build failed: reference 'https://api.rest.sh/schemas/ErrorModel.json' cannot be found at line 11, col 23
```

Given this code linking directly to a JSON Schema file as the reference:

```go
package main

import (
	"github.com/pb33f/libopenapi"
)

var data = `
openapi: "3.1"
paths:
  /fails:
    get:
      responses:
        "200":
          content:
            application/json:
              schema:
                $ref: https://api.rest.sh/schemas/ErrorModel.json
`

func main() {
	doc, err := libopenapi.NewDocument([]byte(data))
	if err != nil {
		panic(err)
	}

	_, errs := doc.BuildV3Model()
	if len(errs) > 0 {
		panic(errs[0].Error())
	}
}
```

As soon as I add a trailing `#` it loads fine. I've traced it down to [FindComponent's HttpResolve branch](https://github.com/pb33f/libopenapi/blob/d2a8379071ea3d85b3a8be2d1f9ae986ee4b4c89/index/spec_index.go#L1574-L1578) which assumes refs will always have a `#` to split on:

```go
case HttpResolve:
	uri := strings.Split(componentId, "#")
	if len(uri) == 2 {
		return index.performExternalLookup(uri, componentId, remoteLookup, parent)
	}
```

This is an issue instead of a PR because I'm not sure of what the correct behavior is. Should we just append a `#` if it is missing and leave the rest of the code as-is or make some deeper changes? It is assumed to be present in a few different places.

I also guess my use-case here is atypical but as Open API 3.1 and JSON Schema have converged it now makes much more sense to link directly to published JSON Schemas, and with the excellent VS Code support for docs/linting as you type when a `$schema` is present I've been publishing my schemas separately for a while now to make live-editing API resources easier.

**Comment by daveshanley** [comment]
This is the only place this check happens (by design) I didn't want to be chasing down remote calls from all over the stack, so that block of code is where it happens. 

The use case is valid, and I didn't think of whole files containing a single reference (I would ban such an activity in my engineering teams); however, it **IS** valid, and I think the code should reflect that choice correctly.

I will look into this today; it needs to be adequately fixed with some testing to validate it.

**Comment by daveshanley** [comment]
After more digging, there is a metric ton of code expecting a component - so to avoid a gigantic refactor that could introduce regressions, I took your suggestion of just simulating an empty component (which then defaults to the root of the doc - which is the desired behavior). 

This fix is available in `v0.3.4`

Let me know how it works for you.

**Comment by danielgtaylor** [comment]
Thanks for the quick fix. I ran into the same thing as you when trying to update the various places that made this assumption. Fix looks fine for my use-cases.

FWIW we aren't hand-writing these files. There is automation that extracts and publishes the JSON Schemas on their own so that other tools that understand JSON Schema but not OpenAPI can use them. I didn't realize one of our teams had directly linked to the schema rather than the OpenAPI with a hash. It was simple enough to fix on our end but I wanted to bring this up here just because it's out of spec. I promise we aren't trying to do horrible things :sweat_smile: 🤣
